### PR TITLE
Validates the MUSICA meta data against the CCPP standard names

### LINKED ISSUE #120: Add test to ensure metadata is correct
Mock host model for testing physics schemes that includes the CCPP code generation step to ensure the metadata is correct

**Acceptance Criteria**

- Test the metadata against the source code
- Add an automated checker that validates the standard names against the source code
- Check the standard names against the official dictionary

**Ideas**
- Use [the functionality](https://github.com/NCAR/ccpp-framework/blob/main/scripts/fortran_tools/offline_check_fortran_vs_metadata.py) in the framework already to test the metadata against the source code.
- See these instructions: https://escomp.github.io/CAM-SIMA-docs/atmospheric_physics/development_workflow/#updating-namesnotindictionarytxt-file
- Use [standard name checker](https://github.com/ESCOMP/CCPPStandardNames/blob/main/tools/meta_stdname_check.py)
- Just trigger warnings on missing standard names

### PR DESCRIPTION
**Pull request by boulderdaze**
Originator(s): @boulderdaze

Summary (include the keyword ['closes', 'fixes', 'resolves'] and issue number):
- Add an automated checker that validates the MUSICA meta data against the CCPP standard names 
- Closes #120 

Describe any changes made to the namelist: N/A

List all files eliminated and why: N/A

List all files added and what they do:  N/A

List all existing files that have been modified, and describe the changes: 
```
M   doc/NamesNotInDictionary.txt
M   schemes/musica/musica_ccpp_namelist.xml
M   test/docker/Dockerfile.musica
M   test/docker/Dockerfile.musica.no_install
M   test/musica/CMakeLists.txt
```
List any test failures: N/A

Is this a science-changing update? New physics package, algorithm change, tuning changes, etc? No

### CODE REVIEW

**Review by mattldawson — APPROVED** [review]
Looks great!

File: test/musica/CMakeLists.txt [hunk 1]
```diff
@@ -75,7 +75,15 @@ add_custom_target(
...
   COMMAND ${Python_EXECUTABLE} ${CMAKE_BINARY_DIR}/../$ENV{CCPP_FORTRAN_TOOLS_PATH}/offline_check_fortran_vs_metadata.py
   --directory ${CMAKE_BINARY_DIR}/metadata_test
+)
+
+# Test metadata against the CCPP standard names
+add_test(
+  NAME test_metadata_against_ccpp_standard_names
+  COMMAND ${Python_EXECUTABLE} ${CMAKE_BINARY_DIR}/../$ENV{CCPP_STD_NAMES_TOOLS_PATH}/meta_stdname_check.py
+  --metafile-loc ${CMAKE_BINARY_DIR}/metadata_test/musica_ccpp.meta
+  --stdname-dict ${CMAKE_CURRENT_SOURCE_DIR}/../../schemes/musica/musica_ccpp_namelist.xml
```
Comment (jimmielin) [inline test/musica/CMakeLists.txt:88]: Just to confirm, is this testing the MUSICA code against a custom MUSICA dictionary of standard names? schemes/musica/musica_ccpp_namelist.xml seems to include a set of name list options and not a standard name dictionary like in the CCPPStandardNames repository (https://github.com/ESCOMP/CCPPStandardNames/blob/main/standard_names.xml). Maybe I misunderstood the `meta_stdname_check.py`, so I'm just confirming this is intended. Thanks!

**Review by jimmielin — CHANGES_REQUESTED** [review]
Thanks @boulderdaze! This seems very useful for MUSICA. I had a question about how the standard name check script is being called and a more general question of if MUSICA is going to maintain its own standard name database or will match the code against the CCPPStandardNames.

File: test/musica/CMakeLists.txt (hunk 1 above)
Comment (boulderdaze) [reply test/musica/CMakeLists.txt:88]: That’s a good question. As far as I know, the current XML file doesn't include any MUSICA standard names yet, and we haven't discussed it so far. The current test primarily checks if the XML formatting is correct and identifies any names that are not in the Standard Name lists. I think this is something we should discuss further. Hi @mattldawson,  do you have any feedback on this test?

```
Non-dictionary standard names found in the following metadata files:
    - air_pressure
    - air_temperature
    - blackbody_temperature_at_surface
    - ccpp_constituent_properties
    - ccpp_constituents
    - ccpp_error_code
    - ccpp_error_message
    - dry_air_density
    - dynamic_constituents_for_musica_ccpp
    - extraterrestrial_radiation_flux
    - geopotential_height_wrt_surface
    - geopotential_height_wrt_surface_at_interface
    - micm_solver_type
    - number_of_grid_cells
    - number_of_photolysis_wavelength_grid_sections
    - photolysis_wavelength_grid_interfaces
    - standard_gravitational_acceleration
    - surface_albedo_due_to_UV_and_VIS_direct
    - surface_geopotential
    - timestep_for_physics
    - vertical_interface_dimension
    - vertical_layer_dimension
```

Comment (mattldawson) [reply test/musica/CMakeLists.txt:88]: I guess I missed that part. I think I agree with @jimmielin - we should be checking against the standard names in `ESCOMP/CCPPStandardNames` in addition to the very few MUSICA-specific standard names (in the `musica_ccpp_namelist.xml`). I think many of the names in the list above should already be in the official list (if we spelled them correctly).

Comment (boulderdaze) [reply test/musica/CMakeLists.txt:88]: Okay, thank you both! I will rework on this.

Comment (boulderdaze) [reply test/musica/CMakeLists.txt]: Here is the updated list of items that are not present in the dictionary. 

```
Non-dictionary standard names found in the following metadata files:
Found in the spreadsheet
    - blackbody_temperature_at_surface
    - extraterrestrial_radiation_flux
    - geopotential_height_wrt_surface_at_interface
    - number_of_photolysis_wavelength_grid_sections
    - photolysis_wavelength_grid_interfaces
    - surface_albedo_due_to_UV_and_VIS_direct

Listed in musica_ccpp_namelist.xml
    - micm_solver_type
    - number_of_grid_cells

Not applicable to either, but there's no need to worry about
    - dynamic_constituents_for_musica_ccpp
```

**Review by jimmielin — APPROVED** [review]
Looks great, thank you for addressing my comments!

**Pull request merged**